apps/meteor/app/api/server/v1/rooms.ts

```diff
@@ -58,3 +96,530 @@ API.v1.addRoute(
 		},
 	},
 );
+
+API.v1.addRoute(
+	'rooms.get',
+	{ authRequired: true },
+	{
+		async get() {
+			const { updatedSince } = this.queryParams;
+
+			let updatedSinceDate;
+			if (updatedSince) {
+				if (isNaN(Date.parse(updatedSince))) {
+					throw new Meteor.Error('error-updatedSince-param-invalid', 'The "updatedSince" query parameter must be a valid date.');
+				} else {
+					updatedSinceDate = new Date(updatedSince);
+				}
+			}
+
+			let result: { update: IRoom[]; remove: IRoom[] } = await Meteor.call('rooms/get', updatedSinceDate);
+
+			if (Array.isArray(result)) {
+				result = {
+					update: result,
+					remove: [],
+				};
+			}
+
+			return API.v1.success({
+				update: result.update.map((room) => this.composeRoomWithLastMessage(room, this.userId)),
```

Comment:
As soon as we have #28432 merged, I'll change this to the created function instead of a "helper" inside the prototype.